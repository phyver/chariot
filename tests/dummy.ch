data unit where Tuple_0 : unit

val f x _ y = x

val g _ x _ = x

val h _ _ _ _ = ()

:show functions

:set verbose 0

:show functions
