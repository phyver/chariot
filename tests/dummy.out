(* ===  functions in environment  =================== *)
val
   f : 'a → 'b → 'c → 'a
    | f x _ y = x

val
   g : 'a → 'b → 'c → 'b
    | g _ x _ = x

val
   h : 'a → 'b → 'c → 'd → unit
    | h _ _ _ _ = ()

(* ================================================== *)

(* ===  functions in environment  =================== *)
val
   f : 'a → 'b → 'c → 'a
    | f x _₁ y = x

val
   g : 'a → 'b → 'c → 'b
    | g _₁ x _₂ = x

val
   h : 'a → 'b → 'c → 'd → unit
    | h _₁ _₂ _₃ _₄ = ()

(* ================================================== *)

